### PR TITLE
[GLUTEN-1780][CH] Add CH backend UT statistic

### PR DESCRIPTION
**Pull request by loneylee**

### CODE REVIEW

File: gluten-core/src/main/scala/io/glutenproject/GlutenPlugin.scala [hunk 1]
```diff
@@ -50,6 +51,11 @@ private[glutenproject] class GlutenDriverPlugin extends DriverPlugin {
 
   override def init(sc: SparkContext, pluginContext: PluginContext): util.Map[String, String] = {
     val conf = pluginContext.conf()
+    if (conf.get("spark.test.home", "") != "") {
```
Comment (zzcclp) [inline gluten-core/src/main/scala/io/glutenproject/GlutenPlugin.scala:54]: equals

File: backends-clickhouse/src/main/scala/io/glutenproject/utils/ExpressionDocUtil.scala [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+import io.glutenproject.expression.ExpressionMappings
+
+import org.apache.spark.sql.catalyst.analysis.FunctionRegistry
+import org.apache.spark.sql.catalyst.expressions.ExpressionInfo
+
+import java.io.{File, PrintWriter}
+import java.util
+
+object ExpressionDocUtil {
```
Comment (rui-mo) [inline backends-clickhouse/src/main/scala/io/glutenproject/utils/ExpressionDocUtil.scala:27]: Please add some comments to explain this class.

File: backends-clickhouse/src/main/scala/io/glutenproject/utils/ExpressionDocUtil.scala [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+    val summary = new StringBuffer()
+    content.append(
+      markdownFormat.format("Group", "Expression", "Spark Class Name", "CH Backend", "Examples"))
+    content.append(markdownFormat.format("----", "----", "----", "----", "----"))
+
+    var total = 0
+    var support = 0
+    group_ordered.forEach(
+      name => {
+        var g_total_cnt = 0
```
Comment (rui-mo) [inline backends-clickhouse/src/main/scala/io/glutenproject/utils/ExpressionDocUtil.scala:58]: Please use camel-case stype for variables. Same for below.

File: gluten-core/src/main/scala/io/glutenproject/GlutenPlugin.scala (hunk 1 above)
Comment (rui-mo) [reply gluten-core/src/main/scala/io/glutenproject/GlutenPlugin.scala:54]: Not all suites set this variable. Do we need to special config for this feature?

**Review by rui-mo — COMMENTED** [review]
Is it possbile to add some tests for the test summary/stats added in this PR?

**Comment by loneylee** [comment]
Run Gluten Clickhouse CI

File: backends-clickhouse/src/main/scala/io/glutenproject/utils/ExpressionDocUtil.scala (hunk 3 above)
Comment (loneylee) [reply backends-clickhouse/src/main/scala/io/glutenproject/utils/ExpressionDocUtil.scala:62]: fix

File: gluten-core/src/main/scala/io/glutenproject/GlutenPlugin.scala (hunk 1 above)
Comment (loneylee) [reply gluten-core/src/main/scala/io/glutenproject/GlutenPlugin.scala]: Add `spark.gluten.sql.ut.statistic` instead.

File: gluten-core/src/main/java/io/glutenproject/test/TestStats.java [hunk 4]
```diff
@@ -17,15 +17,187 @@
...
-  public static boolean offloadGluten = false;
-  public static int suiteTestNumber = 0;
-  public static int offloadGlutenTestNumber = 0;
-
-  public static void reset() {
-    offloadGluten = false;
-    suiteTestNumber = 0;
-    offloadGlutenTestNumber = 0;
-  }
+    private static final String header_format =
```
Comment (rui-mo) [inline gluten-core/src/main/java/io/glutenproject/test/TestStats.java:28]: header_format -> headerFormat?

File: gluten-core/src/main/java/io/glutenproject/test/TestStats.java [hunk 5]
```diff
@@ -17,15 +17,187 @@
...
+            "<tr><th>%s</th><th colspan=5>%s</th></tr>";
+    private static final String row_format =
+            "<tr><td>%s</td><td>%s</td><td>%s</td><td>%s</td><td>%s</td><td>%s</td></tr>";
```
Comment (rui-mo) [inline gluten-core/src/main/java/io/glutenproject/test/TestStats.java:31]: row_format -> rowFormat?

File: gluten-core/src/main/java/io/glutenproject/test/TestStats.java (hunk 4 above)
Comment (loneylee) [reply gluten-core/src/main/java/io/glutenproject/test/TestStats.java]: HEADER_FORMAT

File: gluten-core/src/main/java/io/glutenproject/test/TestStats.java (hunk 5 above)
Comment (loneylee) [reply gluten-core/src/main/java/io/glutenproject/test/TestStats.java:31]: ROW_FORMAT

**Comment by loneylee** [comment]
Run Gluten Clickhouse CI

**Review by baibaichen — APPROVED** [review]
LGTM

**Pull request merged**